src/navigation/wallet/screens/WalletDetails.tsx

```diff
@@ -947,34 +955,43 @@ const WalletDetails: React.FC<WalletDetailsScreenProps> = ({route}) => {
                   <LinkingButtons
                     buy={{
-                      hide: !isCoinSupportedToBuy(
-                        fullWalletObj.currencyAbbreviation,
-                      ),
+                      hide:
+                        !isCoinSupportedToBuy(
+                          fullWalletObj.currencyAbbreviation,
+                        ) ||
+                        (network !== 'livenet' &&
+                          !Object.keys(TestnetFaucets).includes(
+                            currencyAbbreviation,
+                          )),
                       cta: () => {
-                        dispatch(
-                          logSegmentEvent('track', 'Clicked Buy Crypto', {
-                            context: 'WalletDetails',
-                            coin: fullWalletObj.currencyAbbreviation,
-                          }),
-                        );
-                        navigation.navigate('Wallet', {
-                          screen: 'Amount',
-                          params: {
-                            onAmountSelected: async (amount: string) => {
-                              navigation.navigate('BuyCrypto', {
-                                screen: 'BuyCryptoRoot',
-                                params: {
-                                  amount: Number(amount),
-                                  fromWallet: fullWalletObj,
-                                },
-                              });
+                        if (network === 'livenet') {
+                          dispatch(
+                            logSegmentEvent('track', 'Clicked Buy Crypto', {
+                              context: 'WalletDetails',
+                              coin: fullWalletObj.currencyAbbreviation,
+                            }),
+                          );
+                          navigation.navigate('Wallet', {
+                            screen: 'Amount',
+                            params: {
+                              onAmountSelected: async (amount: string) => {
+                                navigation.navigate('BuyCrypto', {
+                                  screen: 'BuyCryptoRoot',
+                                  params: {
+                                    amount: Number(amount),
+                                    fromWallet: fullWalletObj,
+                                  },
+                                });
+                              },
+                              opts: {
+                                hideSendMax: true,
+                                context: 'buyCrypto',
+                              },
                             },
-                            opts: {
-                              hideSendMax: true,
-                              context: 'buyCrypto',
-                            },
-                          },
-                        });
+                          });
+                        } else {
```

Comment:
I would put a warning before leaving the app, saying something like that because it is a testnet wallet you cannot make a purchase, however you will be taken to an external faucet where you can claim test coins.

Reply:
good idea! will add that

Reply:
I was rather suggesting something like this:
![iPhone_12](https://user-images.githubusercontent.com/10983458/180060036-267f2a45-a7ce-4060-8b07-a2f802c87093.jpg)
What do you think?

In addition, let's make sure the faucets are trusted. I would not put redirects to websites that could be a source of scams against our users.

Reply:
added the notification

Reply:
Checking with @JohnathanWhite about faucet links